meteor/client/ui/Status/package-status/PackageStatus.tsx

```diff
@@ -128,108 +63,127 @@ export const PackageStatus = withTranslation()(
 			} else {
 				return 1
 			}
-		}
-		getModifiedHash(onlyRequired: boolean): number {
+		},
+		[props.statuses]
+	)
+	const requiredProgress = useMemo(() => getProgress(true), [props.statuses])
+	const allProgress = useMemo(() => getProgress(false), [props.statuses])
+
+	const getModifiedHash = useCallback(
+		(onlyRequired: boolean): number => {
 			let modifiedHash = 0
-			for (const status of this.props.statuses) {
+			for (const status of props.statuses) {
 				if (onlyRequired && !status.requiredForPlayout) {
 					continue
 				}
 				modifiedHash += status.statusChanged // it's dirty, but it's quick and it works well enough
 			}
 			return modifiedHash
-		}
-		getPackageStatus() {
-			const { t } = this.props
+		},
+		[props.statuses]
+	)
+	const requiredModifiedHash = useMemo(() => getModifiedHash(true), [props.statuses])
+	const allModifiedHash = useMemo(() => getModifiedHash(false), [props.statuses])
 
-			const labelRequiredProgress =
-				this.requiredProgress < 1 ? `${Math.floor(this.requiredProgress * 100)}%` : t('Ready')
-			const labelAllProgress = this.allProgress < 1 ? `${Math.floor(this.allProgress * 100)}%` : t('Done')
+	const requiredProgressLastChanged = useMemo(() => Date.now(), [requiredProgress, requiredModifiedHash])
+	const allProgressLastChanged = useMemo(() => Date.now(), [allProgress, allModifiedHash])
```

Comment:
In the old React docs, this used to be pointed out as an antipattern, since React does not guarantee that the useMemo computation will always be skipped on a render if the deps don't change. However, I don't see that same notice in the new docs, this isn't exactly critical, and It is much more readable than a `useState` and a `useEffect`, though they too could be combined and hidden in a new `useLastChangedTimestamp()` hook.